### PR TITLE
config: Add prettier formatter

### PR DESCRIPTION
**Pull request by TDiazT**
Adding a base setup to have a formatter for the main extension files.
It is not formatting the goal view nor search ui because they have separate config files, which I think can be handled differently (vite setup).
Fixed some typos.

### CODE REVIEW

File: client/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "tabWidth": 2,
```
Comment (TDiazT) [inline client/.prettierrc:2]: Wasn't sure what else to put. This is just an example width (wasn't sure between 2 or 4 🤷 )

File: client/.vscode/settings.json [hunk 2]
```diff
@@ -12,5 +12,7 @@
     "typescript.tsc.autoDetect": "off",
     "ocaml.sandbox": {
         "kind": "global"
-    }
+    },
+    "editor.defaultFormatter": "esbenp.prettier-vscode",
+    "editor.formatOnSave": true,
```
Comment (TDiazT) [inline client/.vscode/settings.json:17]: Question beyond this PR, should these settings be tracked?

File: client/.prettierrc (hunk 1 above)
Comment (rtetley) [reply client/.prettierrc:2]: I think two is fine now ?

File: client/.vscode/settings.json (hunk 2 above)
Comment (rtetley) [reply client/.vscode/settings.json:17]: I actually don't know. My feeling is probably not ?

**Comment by rtetley** [comment]
The CI failures are due to the yarn.nix files not being updated. Could you run the ```nix-shell -p yarn2nix --run "yarn2nix > ` yarn.nix"``` command in the client directory ? That should fix it !
Also sorry I pushed a number of things since then :sweat_smile:
Could you rebase ? I'll then merge !

**Comment by TDiazT** [comment]
Having some 🤷 conflicts, if you could check them @rtetley pls. 
I am having some trouble displaying the goals as well, so I might have broken something trying to rebase.

**Comment by TDiazT** [comment]
> Having some 🤷 conflicts, if you could check them @rtetley pls. I am having some trouble displaying the goals as well, so I might have broken something trying to rebase.

There was a little bug in `extension.ts`. Added a commit renaming the function call (although this one should go on a separate commit probably because it's already in `main`).

**Comment by rtetley** [comment]
> The CI failures are due to the yarn.nix files not being updated. Could you run the `` nix-shell -p yarn2nix --run "yarn2nix > ` yarn.nix" `` command in the client directory ? That should fix it ! Also sorry I pushed a number of things since then 😅 Could you rebase ? I'll then merge !

The CI failures are still due to the yarn.nix files. No need to rebase this time though :-)

**Comment by rtetley** [comment]
Thanks !

**Pull request merged**